app/workers/Cataloger.scala

```diff
@@ -103,70 +73,63 @@ class Cataloger(resmap: ResourceMap, content: StoredContent) {
+            }
+          )
+          idHits = theHits.toSeq
+        } else {
+          val theHits2 = hits map ( h =>
+            h match {
+              case Left(x) => x.attribute.value
+              case Right(x) => concatText(x)
+              // case Right(x) => x.firstChild.get.item().value
+            }
+          )
+          idHits = theHits2.toSeq
+        }
       } else {
-        val theHits2 = hits map ( h =>
-          h match {
-            case Left(x) => x.attribute.value
-            case Right(x) => concatText(x)
-            // case Right(x) => x.firstChild.get.item().value
-          }
-        )
-        idHits = theHits2.toSeq
+        idHits = List()
       }
     }
-  /*
-  idHits = XPath.evaluate(keyParts(0), doc)
-  if (keyParts.length == 2) {
-  // hits need to be transformed by Regex
-  var Transformer = new Regex(keyParts(1))
-  var transList = List[String]()
-  for (hit <- idHits) {
-  var Transformer(x) = hit
-  //var result:String = x
-  transList = x :: transList
-}
-var revList = transList.reverse
-idHits = revList
-*/
-// also stow in infoCache
-  idHits.foreach(println)
-//infoCache += ("id" -> idHits)
-//}
-  val idl = finder.idLabel
-// if idl is an XPath, evaluate it
-  if (idl != null && idl.length > 0 && idl.indexOf("/") >= 0) {
-    //val xpl = new ScalesXPath(idl, Map())
-    lblHits = xpathFind(idl, source)
-    //lblHits =  XPath.evaluate(idl, doc)
-  } else if (idl != null && idl.length > 0) {
-    println("process filtered value; " + filteredValue(idl, 0))
-    var lblList = List[String]()
-    var count = 0
-    for (a <- idHits) {
-      lblList = filteredValue(idl, count) :: lblList
-      count += 1
+    // also stow in infoCache
+    idHits.foreach(println)
+    //infoCache += ("id" -> idHits)
+    if (idHits.size > 0) {
+      val idl = finder.idLabel
+      // if idl is an XPath, evaluate it
+      if (idl != null && idl.length > 0 && idl.indexOf("/") >= 0) {
+        println("in process about to evaluate label: " + idl)
+        lblHits = xpathFind(idl, doc)
+      } else if (idl != null && idl.length > 0) {
+        println("process filtered value; " + filteredValue(idl, 0))
+        var lblList = List[String]()
+        var count = 0
+        for (a <- idHits) {
+          lblList = filteredValue(idl, count) :: lblList
+          count += 1
+        }
+        lblHits = lblList.reverse
+      } else
+        lblHits = List("No label")
+    } else {
+      lblHits = List()
     }
-    //var revLblList = lblList.reverse
-    lblHits = lblList.reverse
-  } else
-    lblHits = List("No label")
-   //}
-   // equalize length if needed
-    if (idHits.length > lblHits.length) {
-      (idHits, lblHits.padTo(idHits.length, "No Label"))
-    } else (idHits, lblHits)
+    // equalize length if needed
+    normalizeHits(idHits, lblHits)
+  }
+
+  private def normalizeHits(hits: Seq[String], lbls: Seq[String]) = {
```

Comment:
I don't have a better answer to this, but if anyone writes some sketchy xpath this could probably mask it for a while. I think we do need that feature where an Analyst can run a Finder against sample data.